dashboard/api.py

```diff
@@ -212,7 +248,43 @@ def get_info_for_course(user, course, user_enrollments, user_certificates):
         status = CourseStatus.OFFERED if next_run is not None else CourseStatus.NOT_OFFERED
         course_data['status'] = status
         if next_run is not None:
-            course_data['runs'].append(format_courserun_for_dashboard(next_run, status))
+            course_data['runs'].append(format_courserun_for_dashboard(next_run, status, position=index))
+            index += 1
+        # add the run of the status anyway if the next run is different from the one just added
+        if next_run is None or run_status.course_run.pk != next_run.pk:
+            course_data['runs'].append(
+                format_courserun_for_dashboard(
+                    run_status.course_run, CourseStatus.NOT_PASSED, position=index)
+            )
+            index += 1
+
+    # add all the other runs with status != NOT_ENROLLED
+    # the first one (or two in some cases) has been added with the logic before
+    for run_status in run_statuses:
+        if run_status.status != CourseRunStatus.NOT_ENROLLED:
+            if (run_status.status == CourseRunStatus.READ_CERT and
+                    user_certificates.has_verified_cert(run_status.course_run.edx_course_key)):
+                # in this case the user might have passed the course also in the past
+                cert = user_certificates.get_verified_cert(run_status.course_run.edx_course_key)
+                course_data['runs'].append(
+                    format_courserun_for_dashboard(
+                        run_status.course_run,
+                        CourseStatus.PASSED,
+                        cert,
+                        position=index
+                    )
+                )
```

Comment:
If we are just updating `index` to keep up with the length of `course_data['runs']` why not have `position=len(course_data['runs']) + 1` for each case here?